newroutes/homepage.js

```diff
@@ -0,0 +1,73 @@
+const autoBind = require("auto-bind")
+const express = require("express")
+
+// Import middleware
+const { BadRequestError } = require("@errors/BadRequestError")
+
+const {
+  attachReadRouteHandlerWrapper,
+  attachWriteRouteHandlerWrapper,
+} = require("@middleware/routeHandler")
+
+const { UpdateHomepageSchema } = require("@validators/RequestSchema")
+
+class HomepageRouter {
+  constructor({ homepagePageService }) {
+    this.homepagePageService = homepagePageService
+    // We need to bind all methods because we don't invoke them from the class directly
+    autoBind(this)
+  }
+
+  // Read homepage index file
+  async readHomepage(req, res) {
+    const { accessToken } = req
+
+    const { siteName } = req.params
+
+    const readResp = await this.homepagePageService.read({
+      siteName,
+      accessToken,
+    })
+
+    return res.status(200).json(readResp)
+  }
+
+  // Update homepage index file
+  async updateHomepage(req, res) {
+    const { accessToken } = req
+
+    const { siteName } = req.params
+    const { error } = UpdateHomepageSchema.validate(req.body, {
+      allowUnknown: true,
+    })
+    if (error) throw new BadRequestError(error)
```

Comment:
if the frontend uses this to display information back to the user, it might be worth writing more fine-grained error messages. 

joi provides [more fine-grained detail](https://joi.dev/api/?v=17.6.0) on the error occurred so i think in any case, we might wanna return that rather than wrapping.

Reply:
Makes sense, https://github.com/isomerpages/isomercms-backend/pull/331/commits/2610f33a177fd7d804a189afb606e5f0229093da